packages/qwik-city/src/runtime/src/link-component.tsx

```diff
@@ -83,11 +85,48 @@ export const Link = component$<LinkProps>((props) => {
       })
     : undefined;
 
-  useVisibleTask$(({ track }) => {
+  // Complements the onQVisible$ event to handle subsequent route navigations
+  useTask$(({ track, cleanup }) => {
     track(() => loc.url.pathname);
-    // Don't prefetch on visible in dev mode
-    if (!isDev && anchorRef.value) {
-      handlePrefetch?.(undefined, anchorRef.value!);
+
+    if (isServer) {
+      return;
+    }
+
+    if (anchorRef.value && handlePrefetch) {
+      const observer = new IntersectionObserver(
+        (entries) => {
+          entries.forEach((entry) => {
```

Comment:
from my works on core in some cases .forEach is even 30% slower than simple for of

Reply:
I'll take it :)